### PR TITLE
trust/extract-jks.c: also honor SOURCE_DATE_EPOCH time

### PR DESCRIPTION
**Pull request by haraldh**
For reproducible builds, accept a define timestamp for the java
keystore.

See https://reproducible-builds.org/docs/source-date-epoch/

### CODE REVIEW

File: trust/extract-jks.c [hunk 1]
```diff
@@ -247,10 +249,38 @@ prepare_jks_buffer (p11_enumerate *ex,
 	 * when this was this certificate was added to the keystore, however
 	 * we don't have that information. Java uses time in milliseconds
 	 */
-	if (_p11_extract_jks_timestamp)
-		now = _p11_extract_jks_timestamp;
-	else
-		now = time (NULL);
+	{
+		char *source_date_epoch;
+		source_date_epoch = getenv("SOURCE_DATE_EPOCH");
```
Comment (ueno) [inline trust/extract-jks.c]: Please use `secure_getenv` (it's not a problem here, but for consistency), and insert a space before the opening parenthesis.

File: trust/extract-jks.c [hunk 2]
```diff
@@ -247,10 +249,38 @@ prepare_jks_buffer (p11_enumerate *ex,
...
+		if (source_date_epoch) {
+			unsigned long long epoch;
+			char *endptr;
+			errno = 0;
+			epoch = strtoull(source_date_epoch, &endptr, 10);
+			if ((errno == ERANGE && (epoch == ULLONG_MAX || epoch == 0))
+			    || (errno != 0 && epoch == 0)) {
+				fprintf(stderr, "Environment variable $SOURCE_DATE_EPOCH: strtoull: %s\n", strerror(errno));
```
Comment (ueno) [inline trust/extract-jks.c]: You could use `p11_message_err`.

File: trust/extract-jks.c [hunk 3]
```diff
@@ -247,10 +249,38 @@ prepare_jks_buffer (p11_enumerate *ex,
...
+				exit(EXIT_FAILURE);
```
Comment (ueno) [inline trust/extract-jks.c]: Although it does the job, I would prefer returning `false`.

**Comment by haraldh** [comment]
This enables reproducible rpm installs

```bash
$ rpm -q --scripts ca-certificates-2018.2.26-2.fc29.noarch
[…]
postinstall scriptlet (using /bin/sh):
[…]
/usr/bin/update-ca-trust

$ fgrep java /usr/bin/update-ca-trust
/usr/bin/p11-kit extract --format=java-cacerts --filter=ca-anchors --overwrite --purpose server-auth $DEST/java/cacerts
```

**Comment by haraldh** [comment]
changed your requested fixups

**Pull request merged**